src/lib/y2partitioner/actions/controllers/blk_device.rb

```diff
@@ -0,0 +1,147 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module Actions
+    module Controllers
+      # This class offers helper methods to perform actions over a block device,
+      # for example, to check if its filesystem exists on disk, it is mounted, etc.
+      class BlkDevice
+        # Block device
+        #
+        # @return [Y2Storage::BlkDevice]
+        attr_reader :device
+
+        # Constructor
+        #
+        # @raise [TypeError] if device is not a block device.
+        #
+        # @param device [Y2Storage::BlkDevice] it has to be a block device,
+        #   (i.e., device#is?(:blk_device) #=> true).
+        def initialize(device)
+          raise(TypeError, "param device has to be a block device") unless device.is?(:blk_device)
+
+          @device = device
+        end
+
+        # Whether the current filesystem of the device exists on the system
+        #
+        # @note The device could have a new filesystem that does not match to the real filesystem
+        #   that actually exists on disk (e.g., after formatting the device).
+        #
+        # @return [Boolean]
+        def committed_current_filesystem?
+          return false unless device.formatted?
+
+          committed_filesystem? &&
+            committed_filesystem.sid == device.filesystem.sid
+        end
+
+        # Whether the filesystem of the device exists on the system and it is mounted
+        #
+        # @note The filesystem on system is checked, independently of it matches to the current
+        #   filesystem of the device.
+        #
+        # @return [Boolean]
+        def mounted_committed_filesystem?
+          return false unless committed_filesystem?
```

Comment:
I think this has one problem. What happen when user modify fs e.g. switch from ext4 to btrfs and this ext4 is mounted? We should offer to unmount it. Do we do?

Reply:
no, now I get it. Name is a bit complex too me :)  Why not name it simple `mounted?` ?

Reply:
meaning that device is somewhere mounted.

Reply:
Probably the terminology is a bit confusing at first, but I prefer to explicitly see "committed" in the method, because it makes pretty clear that it refers to the filesystem in the real system and not the current version in memory.